components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -963,19 +966,17 @@ func (p *pg) CreateRule(ctx context.Context, rule *v2.Rule) (*v2.Rule, error) {
 		return nil, p.processError(err)
 	}
 
-	row := tx.QueryRowContext(ctx,
-		`SELECT query_rule_table_associations($1);`, rule.ID)
-
-	var associations []string
-	if err := row.Scan(pq.Array(&associations)); err != nil {
+	assocMap, err := p.getMapOfRuleAssociations(ctx, tx, rule.ID)
```

Comment:
@blakestier and I have been leaning towards writing all these table lookups as database triggers to be safer. What do you think?

Reply:
Is it safer if the lookup is in a transaction? It's all in a single transaction here.